src/server/utils/dates.py

```diff
@@ -88,63 +89,33 @@ def dates_to_ranges(values: Optional[Sequence[Union[Tuple[int, int], int]]]) ->
-            or (isinstance(values[0], int) and guess_time_value_is_week(values[0])):
+        elif guess_time_value_is_week(first_element):
             return weeks_to_ranges(values)
         else:
             return values
     except:
         return values
 
 def days_to_ranges(values: Sequence[Union[Tuple[int, int], int]]) -> Sequence[Union[Tuple[int, int], int]]:
-    intervals = []
-
-    # populate list of intervals based on original values
-    for v in values:
-        if isinstance(v, int):
-            # 20200101 -> [20200101, 20200101]
-            intervals.append([time_value_to_date(v), time_value_to_date(v)])
-        else: # tuple
-            # (20200101, 20200102) -> [20200101, 20200102]
-            intervals.append([time_value_to_date(v[0]), time_value_to_date(v[1])])
-
-    intervals.sort(key=lambda x: x[0])
-
-    # merge overlapping intervals https://leetcode.com/problems/merge-intervals/
-    merged = []
-    for interval in intervals:
-        # no overlap, append the interval
-        # caveat: we subtract 1 from interval[0] so that contiguous intervals are considered overlapping. i.e. [1, 1], [2, 2] -> [1, 2]
-        if not merged or merged[-1][1] < interval[0] - timedelta(days=1):
-            merged.append(interval)
-        # overlap, merge the current and previous intervals
-        else:
-            merged[-1][1] = max(merged[-1][1], interval[1])
-
-    # convert intervals from dates back to integers
-    ranges = []
-    for m in merged:
-        if m[0] == m[1]:
-            ranges.append(date_to_time_value(m[0]))
-        else:
-            ranges.append((date_to_time_value(m[0]), date_to_time_value(m[1])))
-
-    return ranges
+    return _to_ranges(values, time_value_to_date, date_to_time_value, lambda x: x - timedelta(days=1))
 
 def weeks_to_ranges(values: Sequence[Union[Tuple[int, int], int]]) -> Sequence[Union[Tuple[int, int], int]]:
+    return _to_ranges(values, week_value_to_week, week_to_time_value, lambda x: x - 1)
+
+def _to_ranges(values: Sequence[Union[Tuple[int, int], int]], value_to_date: Callable, date_to_value: Callable, decrement: Callable) -> Sequence[Union[Tuple[int, int], int]]:
     intervals = []
 
-    # populate list of intervals based on original values
+    # populate list of intervals based on original date/week values
     for v in values:
         if isinstance(v, int):
-            # 202001 -> [202001, 202001]
-            intervals.append([week_value_to_week(v), week_value_to_week(v)])
+            # 20200101 -> [20200101, 20200101]
+            intervals.append([value_to_date(v), value_to_date(v)])
         else: # tuple
-            # (202001, 202002) -> [202001, 202002]
-            intervals.append([week_value_to_week(v[0]), week_value_to_week(v[1])])
+            # (20200101, 20200102) -> [20200101, 20200102]
+            intervals.append([value_to_date(v[0]), value_to_date(v[1])])
 
     intervals.sort(key=lambda x: x[0])
```

Comment:
```suggestion
    intervals.sort()
```

providing the sort key here is an unnecessary complication which can confuse the reader.

Reply:
Looked it up and it's indeed the default behavior, good catch!